src/PokemonGymImpl.java

```diff
@@ -1,12 +1,21 @@
 import java.util.List;
+import java.util.Scanner; //Necessary import to resolve 'Cannot resolve 'Scanner''.
+import java.util.Random; //Necessary import to resolve 'Cannot resolve 'Random''.
+import java.util.ArrayList; //Necessary import to resolve 'Cannot resolve 'ArrayList''.
+import java.util.Locale; //Necessary import to resolve 'Cannot resolve 'Locale''.
 
 // Los in deze klasse alle foutmeldingen op door (abstracte) klassen met variabelen en methodes te maken en een interface met methodes (en soms een import).
 public class PokemonGymImpl implements PokemonGym {
 
-
+    private Scanner scanner; //Toevoeging voor Bonusopdracht 3. - declaratie 'Scanner'
     List<Pokemon> pokemons;
 
-    public PokemonGymImpl(List<Pokemon> pokemons) {this.pokemons = pokemons;
+    public PokemonGymImpl() { //Toevoeging voor Bonusopdracht 3. - Initialisatie 'Scanner'
```

Comment:
Je hebt hier 2 constructors. Er kan maar 1 constructor tegelijk worden aangeroepen. De ene consstructor initialiseert de scanner, de ander initialiseert de pokemons List. Je wilt ze eigenlijk allebij initialiseren, maar op deze manier kan dat niet. 
Wat je wel kun doen is om in de list-constructor de scanner constructor aan te roepen met `this()`. Zo kun je beide constructors uitvoeren als je de pokemon lijst in de parameter mee geeft